lib/sqlite/sqlite.go

```diff
@@ -68,6 +70,16 @@ func Open(path string) (*sql.DB, bool, error) {
 		}
 	}
 
+	var foreignKeysEnabled int
+	if err := db.QueryRow("PRAGMA foreign_keys;").Scan(&foreignKeysEnabled); err != nil {
+		return nil, false, xerrors.Errorf("failed to check foreign keys setting: %w", err)
+	}
+	if foreignKeysEnabled == 0 {
+		return nil, false, xerrors.Errorf("foreign keys are not enabled for database [@ %s]", path)
+	}
+
+	log.Infof("Database [@ %s] opened successfully with foreign keys enabled", path)
```

Comment:
are you planning on removing these? any future dbs opened from now on will have it turned on and we're not modifying old dbs so this could be taken for granted, yet?

Reply:
I mean there's no harm with in keeping foreign keys turned on for both future and old DBs. This is a backward compatible change but good to verify that they are turned on because the correctness of the `ChainIndex` depends on it.

Reply:
k, but info log should probably be debug or removed entirely

Reply:
Removed this line completely. Let's resolve this.